frontend/app/src/store/history/actions.ts

```diff
@@ -535,27 +551,97 @@ export const actions: ActionTree<HistoryState, RotkehlchenState> = {
-    } catch (e: any) {
-      const message = i18n
-        .t('actions.ledger_actions.error.description', {
-          error: e.message
-        })
-        .toString();
-      const title = i18n.t('actions.ledger_actions.error.title').toString();
+      setStatus(Status.PARTIALLY_LOADED);
+    };
 
-      await dispatch(
-        'notifications/notify',
-        {
-          title,
-          message: message,
-          severity: Severity.ERROR,
-          display: true
-        } as NotificationPayload,
-        { root: true }
+    const onDefaultError: (message: string) => void = message => {
+      notify(
+        i18n.tc('actions.manual_ledger_actions.error.description', undefined, {
+          error: message
+        }),
+        i18n.tc('actions.manual_ledger_actions.error.title'),
+        Severity.ERROR,
+        true
       );
-    } finally {
-      setStatus(Status.LOADED);
-    }
+    };
+
+    const fetchLocation: (location: TradeLocation) => Promise<void> =
+      async location => {
+        const { taskId } = await api.history.ledgerActions(
+          undefined,
+          undefined,
+          location,
+          source === FETCH_FROM_CACHE
+        );
+        const task = createTask<TaskMeta>(taskId, taskType, {
+          title: i18n.tc('actions.ledger_actions.task.title'),
+          description: i18n.tc(
+            'actions.ledger_actions.task.description',
+            undefined,
+            {
+              exchange: exchangeName(location)
+            }
+          ),
+          ignoreResult: false,
+          numericKeys: [...balanceKeys, 'rate']
+        });
+
+        commit('tasks/add', task, { root: true });
+
+        const { result } = await taskCompletion<
+          LimitedResponse<EntryWithMeta<LedgerAction>>,
+          TaskMeta
+        >(taskType, `${taskId}`);
+
+        const actions = [
+          ...state.ledgerActions.data.filter(
+            action => action.location !== location
+          ),
+          ...result.entries.map(({ entry, ignoredInAccounting }) => ({
+            ...entry,
+            ignoredInAccounting: ignoredInAccounting
+          }))
+        ];
+
+        const data: HistoricData<LedgerActionEntry> = {
+          data: actions,
+          limit: result.entriesLimit,
+          found: result.entriesFound
+        };
+
+        commit(HistoryMutations.SET_LEDGER_ACTIONS, data);
+        setStatus(Status.PARTIALLY_LOADED);
+      };
+
+    const onError: (location: TradeLocation, message: string) => void = (
+      location,
+      message
+    ) => {
+      const exchange = exchangeName(location);
+      notify(
+        i18n.tc('actions.ledger_actions.error.description', undefined, {
+          exchange,
+          error: message
+        }),
+        i18n.tc('actions.ledger_actions.error.title', undefined, { exchange }),
+        Severity.ERROR,
+        true
+      );
+    };
+
+    await Promise.all([
+      fetchDefault().catch(e => onDefaultError(e.message)),
+      (
```

Comment:
So the reason the refresh and action buttons are disabled after loading is that each `fetchLocation` call runs after `setStatus(Status.LOADED)` and resets the status to partially loaded.

One way to avoid that would be just destructuring the array of promises that map returns. By doing this `Promise.all` will wait for the default and all the location promises to complete first.

```suggestion
      ...(
```